node/pkg/p2p/p2p.go

```diff
@@ -268,6 +268,12 @@ func Run(
 
 		logger := supervisor.Logger(ctx)
 
+		var coerr error
+		bootstrapPeers, ccqBootstrapPeers, coerr = checkForCutOver(logger, bootstrapPeers, ccqBootstrapPeers, components)
```

Comment:
might be worth noting here that this side-effects `components.ListeningAddressesPatterns`. in that way, it does a bit more than "check"

Reply:
@evan-gray and I reworked the design to address these issues.